### PR TITLE
implement useCopilotReadable

### PR DESCRIPTION
**Pull request by mme**
Implements `useCopilotReadable`:

- keyword args
- automatic serialization
- custom conversion function
- custom dependencies

### CODE REVIEW

File: CopilotKit/packages/react-core/src/lib/copilot-task.ts [hunk 1]
```diff
@@ -19,26 +19,39 @@ export interface CopilotTaskConfig {
...
   private instructions: string;
   private actions: FrontendAction<any>[];
   private includeCopilotReadable: boolean;
-  private includeCopilotActionable: boolean;
+  private includeCopilotActions: boolean;
 
   constructor(config: CopilotTaskConfig) {
     this.instructions = config.instructions;
     this.actions = config.actions || [];
-    this.includeCopilotReadable = config.includeCopilotReadable || true;
```
Comment (ataibarkai) [inline CopilotKit/packages/react-core/src/lib/copilot-task.ts:36]: whoops, that was a nasty bug, nice find

File: CopilotKit/packages/react-core/src/lib/copilot-task.ts [hunk 2]
```diff
@@ -19,26 +19,39 @@ export interface CopilotTaskConfig {
...
-    this.includeCopilotActionable = config.includeCopilotActionable || true;
+    this.includeCopilotReadable =
+      config.includeCopilotReadable !== undefined ? config.includeCopilotReadable : true;
+    if (config.includeCopilotActions !== undefined) {
```
Comment (ataibarkai) [inline CopilotKit/packages/react-core/src/lib/copilot-task.ts]: a little complex, perhaps we can simplify?
with the right order, no need for else cases etc.

```
// true by default, conditionally override the value if configs are set:
this.includeCopilotActions = true
if(config.includeCopilotActionable !== undefined) {  //
   this.includeCopilotActions = config.includeCopilotActionable
}
if(config.includeCopilotActions !== undefined) { 
    this.includeCopilotActions = config.includeCopilotActions
}
```

File: CopilotKit/packages/react-core/src/hooks/use-make-copilot-readable.ts [hunk 3]
```diff
@@ -2,11 +2,7 @@ import { useContext, useEffect, useRef } from "react";
 import { CopilotContext } from "../context/copilot-context";
 
 /**
- * Adds the given information to the Copilot context to make it readable by Copilot.
- * @param information - The information to be added to the Copilot context.
- * @param parentId - The ID of the parent context, if any.
- * @param categories - An array of categories to control which context are visible where. Particularly useful with CopilotTextarea (see `useMakeAutosuggestionFunction`)
- * @returns The ID of the added context.
+ * @deprecated Use the useCopilotReadable function instead.
```
Comment (ataibarkai) [inline CopilotKit/packages/react-core/src/hooks/use-make-copilot-readable.ts:5]: nice

**Review by ataibarkai — APPROVED** [review]
looks great, just made a minor comment

File: CopilotKit/packages/react-core/src/lib/copilot-task.ts (hunk 2 above)
Comment (mme) [reply CopilotKit/packages/react-core/src/lib/copilot-task.ts]:
```
    this.includeCopilotReadable = config.includeCopilotReadable !== false;
    this.includeCopilotActions =
      config.includeCopilotActions !== false && config.includeCopilotActionable !== false;
 ```

**Pull request merged**